libiocage/lib/Jail.py

```diff
@@ -18,6 +18,49 @@
 
 
 class Jail:
+    """
+    Iocage unit orchestrates a jail's configuration and manages state
+
+    Jails are represented as zfs dataset `zpool/iocage/jails/<NAME>`
+
+    Directory Structure:
+
+        zpool/iocage/jails/<NAME>: The jail's dataset containing it's
+            configuration and root directory. Iocage legacy used to store
+            a jails configuration as ZFS properties on this dataset. Even
+            though the modern JSON config mechanism is preferred.
+
+        zpool/iocage/jails/<NAME>/root: This directory is the dataset
+            used as jail path when starting a jail. Usually the origin
+            or a root dataset is a release root directory
+
+        zpool/iocage/jails/<NAME>/config.json: Jails configured with the latest
+            configuration style store their information in a JSON file. When
+            this file is found in the jail's dataset, libiocage assumes
+            the jail to be a JSON-style jail and ignores other configuration
+            mechanisms
+
+        zpool/iocage/jails/<NAME>/config: Another supported configuration
+            mechanism is a UCL file. It's content is only taken into account if
+            no JSON or ZFS configuration was found
+
+    Jail Types:
+
+        Standalone: The /root dataset gets cloned from a release at creation
+            time. It it not affected by changes to the Release and persists all
+            data within the jail
+
+        NullFS Basejail: The fastest method to spawn a basejail by mounting
+            read-only directories from the release's root dataset by creating
+            a snapshot of the release on each boot of the jail. When a release
+            was updated, the jail is updated as well on the next reboot. This
```

Comment:
`When a release is updated` not was. We should clarify that the jails userland will be updated, but not the configuration in `/etc`. The user would have to check that.